staging/src/k8s.io/apimachinery/pkg/util/httpstream/httpstream.go

```diff
@@ -124,8 +124,8 @@ func Handshake(req *http.Request, w http.ResponseWriter, serverProtocols []strin
 	clientProtocols := req.Header[http.CanonicalHeaderKey(HeaderProtocolVersion)]
 	if len(clientProtocols) == 0 {
 		// Kube 1.0 clients didn't support subprotocol negotiation.
-		// TODO require clientProtocols once Kube 1.0 is no longer supported
-		return "", nil
+		// require clientProtocols once Kube 1.0 is no longer supported
+		return "", fmt.Errorf("clientProtocols is required")
 	}
 
 	if len(serverProtocols) == 0 {
```

Comment:
the same comment is present for `serverProtocols` and was added in the same original commit.
there are probably no tracking issues for these, but it is not clear what happens with `serverProtocols`.

Reply:
The two (client/server) go together. Only 1.0 has no negotiation.